### DISCUSSION TITLE
When Pre-Rendering Routes; Skip if already pre-rendered?

### DISCUSSION THREAD

**Discussion by SleepiestAdam** [nuxt/nuxt#15917] [Questions]
Hi,
I have a site with a lot of pre-rendered pages which is causing deployment to take a long time.

In reality very few of these pages change between deployments.

I was wondering if there was a way to have it  only re-prerender specific pages and/or skip pre-rendering if a pre-rendered version of the given path already exists in the output directory; as this would significantly speed up deployments without impacting the site at all.

The “dream” would be it just identifying if any source files relating to the given path have changed between deployments and conditionally re-rendering based on this, but for now I’d just settle for a manual way of it not having to re-rendering “everything” on each deployment?

Anyone found a way to do this?

**Comment by TheAlexLichter** [comment]
It is not that easy because if you e.g. change content or a component you'd have to figure out which pages "depend" on it and have to change - and based on their change you have to "recursively" check again.

What you can do though is using ISG so sites will be generated on demand via `routeRules` - but right now only netlify and vercel are supported.

**Comment by SleepiestAdam** [comment]
Yeah - to be honest even a flag that (if true) skips the pre-render if the index.html file for the given path already exists in the output directory would be super useful; that way if (for instance) I was just making edits to the independent front-of-site landing page, I could delete that file before the deployment and the other files would be skipped if the index.html in the output folder is present?

With 1000+ pages on our site, it'd significantly speed up deployment.

**Reply by TheAlexLichter** [reply]
I'd say a naive approach could already work by copying the content of the output folder, adapting the routes in a nuxt hook based on what exists in there and what not and render these.
But the above mentioned shortcomings still exist.
